src/dataframe/dataframe.jl

```diff
@@ -596,10 +596,34 @@ Base.setindex!(df::DataFrame, x::Void, col_ind::Int) = delete!(df, col_ind)
 
 Base.empty!(df::DataFrame) = (empty!(df.columns); empty!(index(df)); df)
 
-function Base.insert!(df::DataFrame, col_ind::Int, item::AbstractVector, name::Symbol)
+# TODO: add docstring of make_unique after 0.4 release of DataFrames
+function Base.insert!(df::DataFrame, col_ind::Int, item::AbstractVector, name::Symbol;
+                      make_unique::Bool=false)
     0 < col_ind <= ncol(df) + 1 || throw(BoundsError())
     size(df, 1) == length(item) || size(df, 1) == 0 || error("number of rows does not match")
 
+    if name in _names(df)
```

Comment:
I wonder whether there isn't a more efficient way of checking this by using the `Index` type, which stores a dict mapping column names to their indices.

Reply:
fixed: `name in keys(index(df))` is faster

Reply:
actually `haskey(df, name)` is probably best